tests/acceptance/course-page/complete-second-stage-test.js

```diff
@@ -153,4 +154,50 @@ module('Acceptance | course-page | complete-second-stage', function (hooks) {
     assert.notOk(coursePage.testRunnerCard.isVisible, 'Test runner card disappears');
     assert.ok(coursePage.completedStepNotice.isVisible, 'Completed step notice is visible');
   });
+
+  test('passing stage 2 should show valid clickable stage 2 completion discount', async function (assert) {
+    testScenario(this.server);
+    // The discount timer is currently behind a feature flag
+    const user = signInAsStaff(this.owner, this.server);
+
+    let go = this.server.schema.languages.findBy({ slug: 'go' });
+    let redis = this.server.schema.courses.findBy({ slug: 'redis' });
+
+    let repository = this.server.create('repository', 'withFirstStageCompleted', {
+      course: redis,
+      language: go,
+      user: user,
+    });
+
+    await catalogPage.visit();
+    await catalogPage.clickOnCourse('Build your own Redis');
+
+    this.server.create('submission', 'withStageCompletion', {
+      repository: repository,
+      courseStage: redis.stages.models.sortBy('position')[0],
+    });
+
+    this.server.schema.promotionalDiscounts.create({
+      user: user,
+      type: 'stage_2_completion',
+      percentageOff: 40,
+      expiresAt: new Date(new Date().getTime() + 24 * 60 * 60 * 1000 - 30 * 60 * 1000), // Subtract 30 minutes from the expiration time
+    });
+
+    this.server.create('submission', 'withSuccessStatus', {
+      repository: repository,
+      courseStage: redis.stages.models.sortBy('position')[1],
+    });
+
+    await Promise.all(window.pollerInstances.map((poller) => poller.forcePoll()));
+    await animationsSettled();
+
+    await coursePage.testRunnerCard.clickOnMarkStageAsCompleteButton();
```

Comment:
THe discount creation logic should be placed just before this, that's the "closest" we can get to reality